src/Microsoft.NetCore.Analyzers/UnitTests/Tasks/UseValueTaskCorrectlyTests.cs

```diff
@@ -0,0 +1,1274 @@
+                        // Test0.cs(18,49): warning CA2012: Use ValueTasks correctly.
+                        GetCSharpResultAt(18, 49, UseValueTasksCorrectlyAnalyzer.AccessingIncompleteResultRule)
+                    }
+                );
+        }
+
+        [Fact]
+        public void Diagnostics_UnguardedOutResultAccess()
+        {
+            VerifyCSharp(CSharpBoilerplate(@"
+                using System;
+                using System.Threading.Tasks;
+
+                class C
+                {
+                    public void UnguardedResultAccess(out ValueTask vt0, out ValueTask<string> vt1, out ValueTask<int> vt2, out ValueTask<string> vt3)
+                    {
+                        vt0 = Helpers.ReturnsValueTask();
+                        vt0.GetAwaiter().GetResult();
+
+                        vt1 = Helpers.ReturnsValueTaskOfT<string>();
+                        vt1.GetAwaiter().GetResult();
+
+                        vt2 = Helpers.ReturnsValueTaskOfInt();
+                        vt2.GetAwaiter().GetResult();
+
+                        vt3 = Helpers.ReturnsValueTaskOfT<string>();
+                        _ = vt3.Result;
+                    }
+                }"),
+                    parseOptions: s_csharpPreviewParseOptions,
+                    expected: new DiagnosticResult[]
+                    {
+                        // Test0.cs(9,31): warning CA2012: Use ValueTasks correctly.
+                        GetCSharpResultAt(9, 31, UseValueTasksCorrectlyAnalyzer.AccessingIncompleteResultRule),
+                        // Test0.cs(12,31): warning CA2012: Use ValueTasks correctly.
+                        GetCSharpResultAt(12, 31, UseValueTasksCorrectlyAnalyzer.AccessingIncompleteResultRule),
+                        // Test0.cs(15,31): warning CA2012: Use ValueTasks correctly.
+                        GetCSharpResultAt(15, 31, UseValueTasksCorrectlyAnalyzer.AccessingIncompleteResultRule),
+                        // Test0.cs(18,31): warning CA2012: Use ValueTasks correctly.
+                        GetCSharpResultAt(18, 31, UseValueTasksCorrectlyAnalyzer.AccessingIncompleteResultRule)
+                    }
+                );
+        }
+
+        [Fact]
+        public void Diagnostics_MultipleLocalAwaits()
+        {
+            VerifyCSharp(CSharpBoilerplate(@"
+                using System;
+                using System.Threading.Tasks;
+
+                class C
+                {
+                    public async Task MultipleAwaits()
+                    {
+                        ValueTask vt1 = Helpers.ReturnsValueTask();
+                        await vt1;
+
+                        ValueTask<string> vt2 = Helpers.ReturnsValueTaskOfT<string>();
+                        await vt2;
+
+                        ValueTask<int> vt3 = Helpers.ReturnsValueTaskOfInt();
+                        await vt3;
+
+                        await vt1;
```

Comment:
Can you add a test where awaits are across blocks with differing conditions, like say:

```csharp
if (condition1)
{
   await vt1;
}
else
{
   await vt2;
}

await vt1; // Should be flagged as there is at least one code path with duplicate await
await vt3;

if (condition2)
{
   await vt2; // Should not be flagged as we don't know the relation between condition1 and condition2, so don't know for sure if this a duplicate await

   await vt3;  // Should be flagged as there is at least one code path with duplicate await
}
```